spyderlib/plugins/__init__.py

```diff
@@ -124,6 +217,14 @@ def __init__(self, *args, **kwargs):
         if sys.platform == 'darwin':
             self.setStyleSheet(self.DARWIN_STYLE)
 
+        # Needed for the installation of the event filter
+        self.title = args[0]
+        self.main = args[1]
+        self.dock_tabbar = None
+
+        # Update the QTabBar everytime dockwidget visibility changes
+        self.visibilityChanged.connect(self.install_tab_event_filter)

```

Comment:
Also it even though this operation is triggered often... the filter is only installed once per QTabBar, so it should not impact usability. I hope there was a more intuitive and direct way offered by Qt, but I guess dockwidgets are meant to be used as extra panels in an application with a main  widget.. ;-)